control-plane/connect-inject/consul_dataplane_sidecar.go

```diff
@@ -190,6 +190,60 @@ func (w *MeshWebhook) getContainerSidecarCommand(namespace corev1.Namespace, mpi
 		cmd = append(cmd, fmt.Sprintf("-envoy-admin-bind-port=%d", 19000+mpi.serviceIndex))
 	}
 
+	metricsServer, err := w.MetricsConfig.shouldRunMergedMetricsServer(pod)
+	if err != nil {
+		return nil, fmt.Errorf("unable to determine if merged metrics is enabled: %w", err)
+	}
+	if metricsServer {
```

Comment:
Need tests for this.

Reply:
I migrated the tests over from container_init.go to cover these.